### PR TITLE
[4253] Add filters for start and end academic cycle

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/SsCIMyPQ/4253-m-add-filters-for-start-and-end-academic-cycle

### Changes proposed in this pull request
- Add "End year" filter dropdown
- Change the underlining filter code to use the new `Trainee` relationship `start_academic_year` and `end_academic_year`
- Update `Trainee` factory to set `start_academic_year` and `end_academic_year`

### Guidance to review
- Visit the trainee index page
- Play around with the filters and click into the trainees to make sure the dates match the filter

### Screenshot
Screenshot 2022-06-20 at 15 43 06

### Important business
- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by jhenderson** [comment]
[[M] Add filters for start and end academic cycle](https://trello.com/c/SsCIMyPQ/4253-m-add-filters-for-start-and-end-academic-cycle)

**Review by AbigailMcP — CHANGES_REQUESTED** [review]

File: app/services/trainees/filter.rb [hunk 1]
```diff
@@ -89,6 +89,14 @@ def start_year(trainees, start_year)
       trainees.merge(academic_cycle_scope)
     end
 
+    def end_year(trainees, end_year)
+      return trainees if end_year.blank?
+
+      academic_cycle_scope = AcademicCycle.for_year(end_year).trainees_ending
```
[the remark this answers is not in the document]
Comment (edwardhorsford) [reply app/services/trainees/filter.rb:95]: I'd be happy to use the four digit year in the query string, but we'd then need to map that to the longer string form for the selected filters.

[the remark this answers is not in the document]
Comment (jhenderson) [reply app/services/trainees/filter.rb:95]: You've raised an interesting point. The `end_year` is really expecting the string "2022 to 2023" and not just `2022`. Maybe the term `end_year` is misleading from the backend perspective. @edwardhorsford thinks we should keep the string "2022 to 2023" from a UI perspective, not so fussed about the URL param. I also think we keep the value "2022 to 2023". However, it sounds like the backend needs to make more sense, so I'll address that.

[the remark this answers is not in the document]
Comment (AbigailMcP) [reply app/services/trainees/filter.rb:95]: happy with that!

**Comment by AbigailMcP** [comment]
> @jhenderson can you confirm how we choose which years to show? Is it using start and end years that are available based on the trainees the provider has?

I'm still not sure about this, it looks like both the start and end year filters show the same years, based on whether there are trainees _starting_ in that year.

**Review by edwardhorsford — APPROVED** [review]

**Comment by jhenderson** [comment]
[Seeded records use unrealistic dates](https://trello.com/c/8YggmiFn/4293-seeded-records-use-unrealistic-dates)

**Pull request merged**